components/dashboard/src/App.tsx

```diff
@@ -40,6 +42,33 @@ function App() {
                 setUser(await getGitpodService().server.getLoggedInUser());
             } catch (error) {
                 console.log(error);
+                if (error instanceof ResponseError) {
+                    switch (error.code) {
+                        case ErrorCodes.USER_BLOCKED:
+                            window.location.href = gitpodHostUrl.with({ pathname: "/blocked" }).toString();
+                            break;
+                        case ErrorCodes.SETUP_REQUIRED:
+                            window.location.href = gitpodHostUrl.with({ pathname: "/first-steps" }).toString();
+                            break;
+                        case ErrorCodes.USER_TERMS_ACCEPTANCE_REQUIRED:
+                            window.location.href = gitpodHostUrl.withApi({
+                                pathname: "/tos",
+                                search: `mode=update&returnTo=${encodeURIComponent(window.location.toString())}`,
+                            }).toString();
+                            break;
+                        case ErrorCodes.NOT_AUTHENTICATED:
+                            if (!error.data) {
+                                window.location.href = gitpodHostUrl.withApi({
+                                    pathname: '/login',
+                                    search: 'returnTo=' + encodeURIComponent(window.location.toString()),
+                                }).toString();
+                            }
+                            break;
+                        case ErrorCodes.USER_DELETED:
+                            window.location.href = gitpodHostUrl.asApiLogout().toString();
+                            break;
+                    }
+                }
```

Comment:
@AlexTugarev: Note: These error codes & handling were blindly ported from the old Start page:

https://github.com/gitpod-io/gitpod/blob/5d3f003a20beb516cf4f7434b835a5c8f069eb46/components/dashboard/src/components/start-workspace.tsx#L179-L211

Please double-check whether they all still make sense. Also some pages may be missing (e.g. `/blocked`, `/first-steps`...), and I'm not sure whether these errors can happen when the Start page is loaded inside an iframe (in which case the redirect mechanism should talk to the parent).

Anyway, please let me know if anything more should be done here and I'll be happy to oblige (or, please let me know if you'd prefer to implement this in a separate PR). 🙂